presto-iceberg/src/test/java/com/facebook/presto/iceberg/IcebergDistributedSmokeTestBase.java

```diff
@@ -1660,6 +1661,33 @@ public void testMetadataDeleteOnNonIdentityPartitionColumn(String version, Strin
         metadataDeleteOnTruncateTransform(version, mode, errorMessage);
     }
 
+    @Test
+    public void testNoneAutoCommitTransactionWithFailAndRollback()
+    {
+        Session session = getQueryRunner().getDefaultSession();
+        assertUpdate(session, "CREATE TABLE test_non_autocommit_table (a bigint, b double)");
+        transaction(getQueryRunner().getTransactionManager(), getQueryRunner().getAccessControl())
+                .execute(Session.builder(session)
+                                .setCatalogSessionProperty(ICEBERG_CATALOG, "merge_on_read_enabled", "false")
```

Comment:
I think you can use the `fail` function instead to fail a query?

```sql
SELECT fail('forced failure')
```

Reply:
Thank you, it's a convenient tool to simulate a query failure, fixed!